### DISCUSSION TITLE
Akai MPDs integration - blinking LEDs when a slot is playing

### DISCUSSION THREAD

**Discussion by Brusie** [helgoboss/helgobox#1386] [Q&A]
I'm writing as a coding illiterate, and as a someone who uses Realearn a lot (and Playtime more and more), but using just the features I need for my limited purposes - therefore apologies if I suggest something that's already totally possible, pretty sure with some scripting this shouldn't be too difficult.

I like **Akai MPDs** for they have lots of faders, endless encoders, decent pads for live use, decent customization options. I was trying to set up Playtime to work well with my MPD232, and my preset is already working very well for what I need, but I thought I would really like to find a way for the pad associated with a clip slot that's playing to blink. 
I don't think the MPD ever uses blinking in its limited native possibilities, but I thought it shouldn't be too difficult to send a message to the MPD to swith the led on and off whenever a clip is playing. Tried myself setting up a feedback-only path in Realearn but wasn't successful.

**Comment by helgoboss** [comment]
There are several ways to achieve blinking LEDs if not natively supported by the device. Not sure yet which is the best one in your case.

One way is to use a timer-to-parameter mapping:

1. Create one control-only mapping "Timer"
  - Source = REAPER/Timer
  - Glue Mode = "Toggle"
  - Target = ReaLearn: Compartment parameter (parameter 1)
2. Now create one feedback-only mapping for each LED that should blink
  - Source = whatever MIDI source is necessary to switch your LED on/off
  - Target = ReaLearn: Compartment parameter (parameter 1)

The challenge is to blink only when a clip is playing (I guess?). This is possible using conditional activation but is a bit convoluted. If you find that too complex, let me know. There's also a second way I can think of.

**Reply by Brusie** [reply]
Thank you very much, Benjamin, I managed to make it work! For whoever may find themselves in the same situation: I basically used a second compartment parameter in order to handle the conditional activation. 
I'll need to work some more in order to handle the blinking with respect to the "exclusive mode" per column, but it's going to be fun!
